test/unit/lib/plugins/aws/package/compile/events/apiGateway/lib/validate.test.js

```diff
@@ -10,6 +11,22 @@ const ServerlessError = require('../../../../../../../../../../lib/serverless-er
 describe('#validate()', () => {
   let serverless;
   let awsCompileApigEvents;
+  let validated;
+
+  const runServerless_ = async (options, runValidate = true) => {
+    const result = await runServerless({
+      fixture: 'function',
+      command: 'package',
+      lastLifecycleHookName: 'before:package:compileEvents',
+      ...options,
+    });
+
+    serverless = result.serverless;
+    awsCompileApigEvents = serverless.pluginManager.getPluginByClass(AwsCompileApigEvents);
```

Comment:
I would avoid this refactoring - I think we should just configure test with `runServerless` for these new default values and confirm on the resulting CloudFormation template the outputs and do not run internal methods explicitly. We don't want to rely on internal methods in tests in any way because it makes refactoring harder in the future (we've already been bitten by it multiple times).

Reply:
Thanks for the storytelling! I can imagine now what you guys have experienced. Resolved!